src/strategy/redis_cache_provider.ts

```diff
@@ -62,80 +67,109 @@ export const getExtendedRedisCacheProvider = (
+            ),
+            TE.map(idpIssuer => ({ idpIssuer, AuthnRequestID }))
+          )
+        ),
+        TE.chain(_ => {
           const v: SAMLRequestCacheItem = {
             RequestXML,
             createdAt: new Date(),
             idpIssuer: _.idpIssuer
           };
-          return taskify(
-            (
-              key: string,
-              data: string,
-              flag: "EX",
-              expiration: number,
-              callback: (err: Error | null, value: unknown) => void
-            ) => redisClient.set(key, data, flag, expiration, callback)
-          )(
-            `${keyPrefix}${_.AuthnRequestID}`,
-            JSON.stringify(v),
-            "EX",
-            keyExpirationPeriodSeconds
-          )
-            .mapLeft(
+          return pipe(
+            TE.taskify(
+              (
+                key: string,
+                data: string,
+                flag: "EX",
+                expiration: number,
+                callback: (err: Error | null, value: unknown) => void
+              ) => redisClient.set(key, data, flag, expiration, callback)
+            )(
+              `${keyPrefix}${_.AuthnRequestID}`,
+              JSON.stringify(v),
+              "EX",
+              keyExpirationPeriodSeconds
+            ),
+            TE.mapLeft(
               err =>
                 new Error(`SAML#ExtendedRedisCacheProvider: set() error ${err}`)
-            )
-            .map(() => v);
-        });
+            ),
+            TE.map(() => v)
+          );
+        })
+      );
     },
     get(AuthnRequestID: string): TaskEither<Error, SAMLRequestCacheItem> {
-      return taskify(
-        (key: string, callback: (e: Error | null, value?: string) => void) => {
-          redisClient.get(key, callback);
-        }
-      )(`${keyPrefix}${AuthnRequestID}`)
-        .mapLeft(
+      return pipe(
+        TE.taskify(
+          (
+            key: string,
+            callback: (e: Error | null, value?: string) => void
+          ) => {
+            redisClient.get(key, (e, v) =>
+              // redis callbacks consider empty value as null instead of undefined,
+              //  hence the need for the following wrapper to convert nulls to undefined
+              callback(e, v === null ? undefined : v)
+            );
+          }
+        )(`${keyPrefix}${AuthnRequestID}`),
+        TE.mapLeft(
           err =>
             new Error(`SAML#ExtendedRedisCacheProvider: get() error ${err}`)
-        )
-        .chain(value =>
-          fromEither(
-            parseJSON(value, toError).chain(_ =>
-              SAMLRequestCacheItem.decode(_).mapLeft(
-                __ =>
-                  new Error(
-                    `SAML#ExtendedRedisCacheProvider: get() error ${readableReport(
-                      __
-                    )}`
+        ),
+        TE.chain(value =>
+          TE.fromEither(
+            pipe(
+              E.parseJSON(value, E.toError),
```

Comment:
use `parse` from `fp-ts/lib/Json`  instead, `E.parseJson` is deprecated
```suggestion
              parse(value),
              E.mapLeft(E.toError),
```